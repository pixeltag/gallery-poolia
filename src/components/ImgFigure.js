import { useCallback, useEffect } from "react";

const ImgFigure = ({ title, src, full, description, openModalCallback }) => {
  const handleOpenModal = () => {
    openModalCallback({ full, description, title });
  };

  const enterPressHandler = useCallback((e) => {
    if (e.keyCode === 13) {
      document.activeElement.click();
    }
  }, []);

  useEffect(() => {
    document.addEventListener("keydown", enterPressHandler, false);
  }, [enterPressHandler]);

  return (
    <figure
      className="img-figure"
      onClick={() => handleOpenModal()}
      tabIndex="0"
    >
      <img src={src} alt={title} />
      <figcaption className="title-figure">
        <h3>{title ? title : "Image title"}</h3>
      </figcaption>
    </figure>
  );
};

export default ImgFigure;
