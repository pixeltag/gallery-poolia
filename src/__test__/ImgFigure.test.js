import { render, cleanup } from "@testing-library/react";
import { unmountComponentAtNode } from "react-dom";

import { shallow, configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

import ImgFigure from "../components/ImgFigure";

configure({ adapter: new Adapter() });

let container = null;
beforeEach(() => {
  // setup a DOM element as a render target
  container = document.createElement("div");
  document.body.appendChild(container);
});

afterEach(() => {
  // cleanup on exiting
  unmountComponentAtNode(container);
  cleanup();
  container.remove();
  container = null;
});

const fakeImage = {
  title: "image titile",
  src: "url-src",
  full: "url-full-src",
  description: "image description",
};

test("<ImgFigure /> component renders correctly", async () => {
  const container = render(<ImgFigure />);
  const figureElement = await container.findByRole("figure");
  expect(figureElement).toBeInTheDocument();

  const titleElement = await container.findByTestId("title");
  expect(titleElement.textContent).toBe("Image title");
});

test("<ImgFigure /> component renders image data", async () => {
  const container = render(<ImgFigure {...fakeImage} />);
  const imgElement = await container.findByRole("img");
  expect(imgElement.getAttribute("src")).toBe(fakeImage.src);
  expect(imgElement.getAttribute("alt")).toBe(fakeImage.title);
});

test("<ImgFigure /> component renders image metadata title", async () => {
  const container = render(<ImgFigure {...fakeImage} />);
  const titleElement = await container.findByTestId("title");
  expect(titleElement.textContent).toBe(fakeImage.title);
});

test("<ImgFigure /> click the figure fire function callback", async () => {
  const openModelMocked = jest.fn();
  const container = shallow(
    <ImgFigure {...fakeImage} openModalCallback={openModelMocked} />
  );
  const figureElement = await container.find("figure");
  figureElement.simulate("click");
  expect(openModelMocked.mock.calls.length).toEqual(1);
});

test("Gallery component should match the snapshot", () => {
  const wrapper = shallow(<ImgFigure />);
  expect(wrapper.html()).toMatchSnapshot();
  cleanup();
});
