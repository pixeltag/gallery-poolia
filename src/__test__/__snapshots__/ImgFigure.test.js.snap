// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gallery component should match the snapshot 1`] = `"<figure class=\\"img-figure\\" tabindex=\\"0\\"><img/><figcaption class=\\"title-figure\\"><h3 data-testid=\\"title\\">Image title</h3></figcaption></figure>"`;
